x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/googleaistudio/GoogleAiStudioService.java

```diff
@@ -287,10 +312,8 @@ protected void doChunkedInfer(
                 EmbeddingRequestChunker.EmbeddingType.FLOAT
             ).batchRequestsWithListeners(listener);
         }
-
         for (var request : batchedRequests) {
-            var action = googleAiStudioModel.accept(actionCreator, taskSettings, inputType);
-            action.execute(new DocumentsOnlyInput(request.batch().inputs()), timeout, request.listener());
+            doInfer(model, new DocumentsOnlyInput(request.batch().inputs()), taskSettings, inputType, timeout, request.listener());
```

Comment:
Why do we need to run `doInfer` on the return from the embedding request chunker? Shouldn't the embedding request chunker be handling this for us already?

Reply:
I think the embedding request chunker is just splitting up the request into N requests and then providing a merging method when the listener is called.  The actual thing calling model inference is a combination of the visitor returned by the `accept` method and then an `execute` on that visitor.  Since that visitor provides the forking between model extensions, and `doInfer` now does that, I figured we could just call the `doInfer` method with the batch